src/cmdrun.rs

```diff
@@ -64,36 +74,81 @@ impl CmdRun {
 
     // This method is public for regression tests
     pub fn run_on_content(content: &str, working_dir: &str) -> Result<String> {
-        let re = Regex::new(r"<!--[ ]*cmdrun (.*)-->\n").unwrap();
         let mut err = None;
 
-        let content = re
+        let mut result = CMDRUN_REG_NEWLINE
             .replace_all(content, |caps: &Captures| {
-                match CmdRun::run_cmdrun(caps[1].to_string(), working_dir) {
-                    Ok(s) => s,
-                    Err(e) => {
-                        err = Some(e);
-                        String::new()
-                    }
-                }
+                Self::run_cmdrun(caps[1].to_string(), working_dir, false).unwrap_or_else(|e| {
+                    err = Some(e);
+                    String::new()
+                })
+            })
+            .to_string();
+
+        if let Some(e) = err {
+            return Err(e);
+        }
+
+        result = CMDRUN_REG_INLINE
+            .replace_all(result.as_str(), |caps: &Captures| {
+                Self::run_cmdrun(caps[1].to_string(), working_dir, true).unwrap_or_else(|e| {
+                    err = Some(e);
+                    String::new()
+                })
             })
             .to_string();
 
         match err {
-            None => Ok(content),
+            None => Ok(result),
             Some(err) => Err(err),
         }
     }
 
+    #[cfg(target_family = "windows")]
+    fn correct_linebreaks(str: Cow<'_, str>, inline: bool) -> String {
+        let mut res = String::with_capacity(str.len());
+        match inline {
+            true => {
+                let str = str.trim_end();
+                let count = str.lines().count();
+                for (i, line) in str.lines().enumerate() {
+                    res.push_str(line);
+                    if i < count - 1 {
+                        res.push_str("\r\n");
+                    }
+                }
+
+                res
+            }
+            false => {
+                for line in str.lines() {
+                    res.push_str(line);
+                    res.push_str("\r\n");
+                }
+
+                res
+            }
+        }
```

Comment:
Not sure myself if it's even necessary, I was working backwards from trying to get the regression tests to work.
Some commands like `yes` output unix-style linebreaks which don't start with`\r` and that caused the tests to break. In practice, a user wouldn't notice the difference, but I suppose there is value in having consistency? Either way your solution is way cleaner.

Reply:
Theoretically someone could be using a tool to verify the resulting markdown and not be able to easily validate it because of inconsistent line breaks, I suppose

Reply:
This is interesting ahah. Thank you for the fix!